src/vehicle.cpp

```diff
@@ -1460,13 +1460,14 @@ bool vehicle::start_engine( const int e )
 
 void vehicle::start_engines( const bool take_control )
 {
-    int has_engine = false;
     int start_time = 0;
 
-    // if no engines enabled then enable all before trying to start the vehicle
-    if( std::none_of( engines.begin(), engines.end(), [&]( int idx ) {
+    int has_engine = std::count_if( engines.begin(), engines.end(), [&]( int idx ) {
```

Comment:
The name suggests `bool`. I'd replace it with `engine_count` or `num_of_engines`.